js/stdlib/ts/ETH_like_compiled.ts

```diff
@@ -140,15 +141,17 @@ const T_Bytes = (len:number): ETH_Ty<CBR_Bytes, ETH_Bytes> => {
   const me = {
     ...CBR.BT_Bytes(len),
     munge: ((bv: CBR_Bytes): ETH_Bytes => {
-      const bs = Array.from(ethers.utils.toUtf8Bytes(bv));
+      const bs = Array.from(canonicalToBytes(bv));
       return (len <= byteChunkSize) ? bs : splitToChunks(bs, byteChunkSize);
     }),
     unmunge: ((nvs: ETH_Bytes): CBR_Bytes => {
-      const go = (nv:any) => hexToString(ethers.utils.hexlify(unBigInt(nv)))
-      const nvs_s = (len <= byteChunkSize) ? go(nvs) : nvs.map(go);
-      const nvss = "".concat(...nvs_s);
-      // debug(me.name, nvs, nvss);
-      return me.canonicalize(nvss);
+      const go = (nv:any, i: number) => {
+        const r = ethers.utils.hexlify(unBigInt(nv));
+        return (i > 0 && r.startsWith('0x')) ? r.slice(2) : r;
```

Comment:
This `0x` thing here makes this function very different than before and it doesn't seem to acknowledge the ui8a vs not thing. Maybe a better way to implement this function is to turn everything into ui8a uniformly and then map or not map?

Reply:
What do you mean by acknowledge the ui8a vs not? It will still produce ui8a's if the bytes are not utf-8 via the `hexToString` call below. I kind of just changed the order of operations from: calling `hexToString` on every element and joining => joining every element and calling `hexToString`. This is to ensure that every element of an array is of the same type, e.g. so we don't get `[string, string, ui8a, string]` for a long `bytes` value that isn't utf-8

Reply:
Okay, I think I understand now